projects/libyuv/libyuv_fuzzer.cc

```diff
@@ -0,0 +1,78 @@
+/* # Copyright 2020 Google Inc. */
+/* # */
+/* # Licensed under the Apache License, Version 2.0 (the "License"); */
+/* # you may not use this file except in compliance with the License. */
+/* # You may obtain a copy of the License at */
+/* # */
+/* #      http://www.apache.org/licenses/LICENSE-2.0 */
+/* # */
+/* # Unless required by applicable law or agreed to in writing, software */
+/* # distributed under the License is distributed on an "AS IS" BASIS, */
+/* # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. */
+/* # See the License for the specific language governing permissions and */
+/* # limitations under the License. */
+/* # */
+/* ################################################################################ */
+
+#include <iostream>
+#include <vector>
+#include <climits>
+
+#include <fuzzer/FuzzedDataProvider.h>
+#include "libyuv.h"
+#include "fuzz_common.h"
+
+using namespace std;
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+    if (size == 0)
+    {
+        libyuv::I420ToNV21(data, 0, data, 0, data, 0, nullptr, 0, nullptr, 0, 0, 0);
+        return 0;
+    }
+
+    FuzzedDataProvider fuzzed_data(data, size);
+
+    int range_size = (static_cast<int>(size) + 1) / 2;
+    int width = 0, height = 0, kPixels = 0, kHalfPixels = 0;
+    do
+    {
+        height = fuzzed_data.ConsumeIntegralInRange(0, range_size);
+        width = fuzzed_data.ConsumeIntegralInRange(0, range_size) ;
+        kPixels = width * height;
+        kHalfPixels = (width + 1) / 2 * (height + 1) / 2;
+    }
+    while (kPixels + kHalfPixels * 2 > size);
+
+    height = fuzzed_data.ConsumeBool() ? height : -height;
+
+    int random_num = fuzzed_data.ConsumeIntegralInRange(0, INT_MAX);
+    int cpu_flags = generate_cpuflags(random_num);
+    write_conf(width, height, cpu_flags);
+
+    std::vector<uint8_t> first_part =
+        fuzzed_data.ConsumeBytes<uint8_t>(kPixels);
+    std::vector<uint8_t> second_part =
+        fuzzed_data.ConsumeBytes<uint8_t>(kHalfPixels);
+    std::vector<uint8_t> third_part =
+        fuzzed_data.ConsumeRemainingBytes<uint8_t>();
+
+    uint8_t *dst_data = new uint8_t [size];
+
+    uint8_t *dst_y_data = dst_data;
+    uint8_t *dst_uv_data = dst_data + kPixels;
+
+    const uint8_t *src_y_data = first_part.data();
+    const uint8_t *src_u_data = second_part.data();
+    const uint8_t *src_v_data = third_part.data();
+
+    int stride = width;
+    width = fuzzed_data.ConsumeIntegralInRange(0, width);
+
+    libyuv::I420ToNV21(src_y_data, stride, src_u_data, (stride + 1) / 2, src_v_data, (stride + 1) / 2, dst_y_data, stride, dst_uv_data, ((width + 1) / 2) * 2, width, height);
```

Comment:
(stride + 1) / 2 is misleading.
a stride is how many bytes to advance from row to row.
You could compute the stride based on width or come up with some stride variables.
src_y_stride = width;
src_uv_stride = (width + 1) / 2;
dst_y_stride = width;
dst_uv_stride = ((width + 1) / 2) * 2;

The dst uv stride you didn't round.  if the width is odd, the stride needs to be rounded up.
e.g. width =3
dst_uv_stride is 4.